rsp/runtimes/bundles/org.jboss.tools.rsp.server.generic/src/main/java/org/jboss/tools/rsp/server/generic/servertype/GenericServerBehavior.java

```diff
@@ -221,7 +236,7 @@ protected IPublishControllerWithOptions getPublishController() {
 			JSONMemento publishMemento = behaviorMemento.getChild("publish");
 			String deployPath = publishMemento.getString("deployPath");
 			String approvedSuffixes = publishMemento.getString("approvedSuffixes");
-			String[] suffixes = approvedSuffixes == null ? null : approvedSuffixes.split(",");
+			String[] suffixes = approvedSuffixes == null ? null : approvedSuffixes.split(",", -1);
```

Comment:
By reading the documentation i think it should be approvedSuffixes.split(",", 0); no? Otherwise the string "war," will create an array with two elements (the second empty)

Reply:
That's the intention. The intention is to allow a server to declare support for no-suffix... which is important for tomcat since exploded deployments will not have a suffix.